finality-aleph/src/base_protocol/config.rs

```diff
@@ -0,0 +1,54 @@
+use std::iter;
+
+use array_bytes::bytes2hex;
+use sc_network::{
+    config::{NonDefaultSetConfig, NonReservedPeerMode, NotificationHandshake, Role, SetConfig},
+    NotificationService,
+};
+use sc_network_common::sync::message::BlockAnnouncesHandshake;
+use sp_core::H256;
+use sp_runtime::traits::{Block, Header};
+
+use crate::{BlockHash, BlockNumber};
+
+/// Generate a config for the base protocol and the notification service that should be passed to its service.
+pub fn generate_config<B>(
```

Comment:
Find a better name for this function - this does not only create config

Reply:
Ok, why I have problem with that: 
- normally when Im reading function `generate_smth` I expect it to create `smth`. Extra stuff from such function is misleading, especially when its Boxed trait ;d
- I feel like this funciton is acually more about protocol service rather than just config, even though the service is a side effect of the ::new.

My propositions:
* `setup_base_protocol_service`
* `setup_protocol_service`
* `configure[_base]_protocol_service`

Reply:
I strongly oppose calling it `service`, since this has a completely different meaning in our codebase, unfortunately. x< How about `setup_base_protocol`? Or perhaps just `setup` or `configure` (with a slight preference for the latter from me), since they are already in `base_protocol`? If we want to stress that we are also returning that interface maybe we could incorporate that word somehow, although I don't see how.

Reply:
just setup is enough - since it will be imported form the `base_protocol`

Reply:
Renamed then.